fql/setup.js

```diff
@@ -0,0 +1,148 @@
+// ---------------------- Step 1: Create a "Users" collection ----------------------
+CreateCollection({ name: 'Users' });
+
+// ---------------------- Step 2: Create "Canvas" collection ----------------------
+CreateCollection({ name: 'Canvas' });
+
+// ---------------------- Step 3: Create indexes ----------------------
+// Index to filter users by email
+// Necessary for authentication, to find the user document based on their email
+CreateIndex({
+  name: 'users_by_email',
+  source: Collection('Users'),
+  terms: [
+    { field: ['data', 'email'] },
+  ],
+  unique: true,
+});
+
+// Index to filter canvas by owner
+// Necessary for real-time subscription, to retrieve the canvas of the current user
+CreateIndex({
+  name: 'canvas_by_owner',
+  source: Collection('Canvas'),
+  // Needs permission to read the Users, because "owner" is specified in the "terms" and is a Ref to the "Users" collection
+  permissions: { read: Collection('Users') },
```

Comment:
here you are mixing the legacy permissions sub-system and bellow you are using the new ABAC roles, we don't recommend mixing both systems.

Reply:
Thanks for your feedback, I wasn't aware I was mixing both systems.

I only did that because without it the index doesn't work, but I don't really understand what I should do instead using ABAC. I basically followed what someone else told me to and it worked well.

I understand anything that's not using a `Query` is using the old permissions sub-system? Or am I mistaken? (how do you tell what's ABAC and what isn't?)

Reply:
ABAC will be evaluated every time you use a token, all roles membership will be evaluated to detect which one contains a resource collection of the same type of your instance in the token, after this the privileges will be evaluated, I saw you have a `Editor` role that contains the collection `Users` on membership.

Reply:
Okay, what I want is to allow authenticated users to use the index `canvas_by_owner`. It's used there: https://github.com/Vadorequest/rwa-faunadb-reaflow-nextjs-magic/blob/7625102c5df7a6b449589e4b9be2a3ce1744240c/src/utils/canvasStream.ts#L155-L160

Only the editors have the right to use the `canvas_by_owner` index.

I don't really understand what I should change, I don't see any issue with the current setup, it seems secure to me. 🤔